### ISSUE TITLE
CUDA ERROR

### ISSUE THREAD

**Issue by dephaniehe** [AlexeyAB/darknet#3395]
When I tried to use zed camera and run the script darknet.py from zed-yolo on Windows, it shows the following CUDA error:

CUDA status Error: file: C:/Project/zed-yolo/zed-yolo-master/libdarknet/src/convolutional_kernels.cu : cuda_convert_f32_to_f16() : line: 137 : build time: May 18 2019 - 14:03:35
CUDA Error: invalid resource handle

But I already successfully built with Cmake. Is there any solution to this?

**Comment by AlexeyAB** [comment]
Can  you show output of commands?
```
nvcc --version
nvidia-smi
```

What CUDA and cuDNN versions do you use?

**Comment by dephaniehe** [comment]
I am using CUDA 10.0 and cuDnn v7.5.1. Here is the output of nvcc version:

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2018 NVIDIA Corporation
Built on Sat_Aug_25_21:08:04_Central_Daylight_Time_2018
Cuda compilation tools, release 10.0, V10.0.130


And here is the output of nvidia-smi:

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 411.70                 Driver Version: 411.70                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name            TCC/WDDM | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce RTX 208... WDDM  | 00000000:01:00.0  On |                  N/A |
| 41%   36C    P8    22W / 260W |    988MiB / 11264MiB |      1%      Default |
+-------------------------------+----------------------+----------------------+
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1072    C+G   Insufficient Permissions                   N/A      |
|    0      1272    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0      1816    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0      1876    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0      3392    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0      4420    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0      4612    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0      5252    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0      5316    C+G   ...t_cw5n1h2txyewy\ShellExperienceHost.exe N/A      |
|    0      5412    C+G   C:\Windows\explorer.exe                    N/A      |
|    0      5436    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0      6176    C+G   ...dows.Cortana_cw5n1h2txyewy\SearchUI.exe N/A      |
|    0      6300    C+G   ...o\2017\Community\Common7\IDE\devenv.exe N/A      |
|    0      6672    C+G   ...oftEdge_8wekyb3d8bbwe\MicrosoftEdge.exe N/A      |
|    0      6760    C+G   C:\Program Files\Unity\Editor\Unity.exe    N/A      |
|    0      6972    C+G   ...mmersiveControlPanel\SystemSettings.exe N/A      |
|    0      7272    C+G   ...46.60.0_x64__kzf8qxf38zg5c\SkypeApp.exe N/A      |
|    0      7296    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0      7316    C+G   ....545.0_x64__8wekyb3d8bbwe\YourPhone.exe N/A      |
|    0      7988    C+G   ...ge_8wekyb3d8bbwe\MicrosoftPdfReader.exe N/A      |
|    0      8060    C+G   ...gram Files\Unity\Editor\UnityHelper.exe N/A      |
|    0      8112    C+G   ...osoft.LockApp_cw5n1h2txyewy\LockApp.exe N/A      |
|    0      8204    C+G   ...4.0_x64__8wekyb3d8bbwe\WinStore.App.exe N/A      |
|    0      8464    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0      8884    C+G   ...hell.Experiences.TextInput.InputApp.exe N/A      |
|    0      8920    C+G   ...ost.CLR.x86\ServiceHub.Host.CLR.x86.exe N/A      |
|    0      9040    C+G   ...x64__8wekyb3d8bbwe\Microsoft.Photos.exe N/A      |
|    0      9240    C+G   C:\Windows\System32\MicrosoftEdgeSH.exe    N/A      |
|    0      9308    C+G   ...ge_8wekyb3d8bbwe\MicrosoftPdfReader.exe N/A      |
|    0     10456    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     10500    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     11128    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     11372    C+G   ...11411.0_x64__8wekyb3d8bbwe\Video.UI.exe N/A      |
|    0     11584    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     11656    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     12084    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     12224    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     12388    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     13044    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     14100    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     14556    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     14652    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     14708    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     16524    C+G   ...o\2019\Community\Common7\IDE\devenv.exe N/A      |
|    0     16628    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     16692    C+G   ...o\2019\Community\Common7\IDE\devenv.exe N/A      |
|    0     17416    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     19168    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     19900    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     20496    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     22240    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
|    0     23092    C+G   ..._x64__8wekyb3d8bbwe\Microsoft.Notes.exe N/A      |
|    0     24464    C+G   C:\Windows\System32\MicrosoftEdgeCP.exe    N/A      |
+-----------------------------------------------------------------------------+

**Comment by AlexeyAB** [comment]
Do you use repository https://github.com/stereolabs/zed-yolo ?

Can you successfully build this https://github.com/AlexeyAB/darknet repository  
how to build https://github.com/AlexeyAB/darknet#how-to-compile-on-linux
and run this command
`LD_LIBRARY_PATH=./:$LD_LIBRARY_PATH ./uselib data/coco.names cfg/yolov3.cfg yolov3.weights zed_camera`

**Comment by dephaniehe** [comment]
Yes, I use the repository [https://github.com/stereolabs/zed-yolo](url). 
I can build the https://github.com/AlexeyAB/darknet repository with CMake and successfully run the command `uselib.exe data/coco.names cfg/yolov3.cfg yolov3.weights zed_camera`.

**Comment by cenit** [comment]
@dephaniehe then maybe it’s a problem more related to that repository? @adujardin do you think it would be feasible to simplify the zed-yolo repo and just use this one as a submodule or better just the library as a dependency, linking to it, in order to adopt quicker fixes and enhancements?

**Comment by adujardin** [comment]
@cenit Yes, it would be better. The idea was to facilitate the setup, especially for the python sample which requires a specific path.
I don't have time right now, but I'll update it later.

**Comment by cenit** [comment]
of course for any problem using darknet as an external library please let me know, your sdk would be a terrific public test for it. I think you'll need also a decent api surface enhacement if you were used to just building darknet inside your project.

**Comment by dephaniehe** [comment]
> @dephaniehe then maybe it’s a problem more related to that repository? @adujardin do you think it would be feasible to simplify the zed-yolo repo and just use this one as a submodule or better just the library as a dependency, linking to it, in order to adopt quicker fixes and enhancements?

Also, I am using the GeForce RTX 2080Ti, will the floating operation affect the results?
